p2p/trust/store.go

```diff
@@ -135,9 +138,27 @@ func (tms *TrustMetricStore) SaveToDB() {
 
 /* Private methods */
 
+// addPeerTrustMetric takes an existing trust metric and associates it with a peer and reactor ID.
+// This version of the method assumes the mutex has already been acquired
+func (tms *TrustMetricStore) addPeerTrustMetric(peer, reactor string, tm *TrustMetric) {
+	if peer == "" || reactor == "" || tm == nil {
```

Comment:
not of a fan of this defensive programming. when tm can be `nil`? when peer == ''?

Reply:
what don't you like? possibly masks bugs?

Reply:
exactly because of my questions. it creates additional flow or branches of execution (not sure how to explain this), but every new dev will be asking him/herself these questions: when tm can be nil? when peer == ''? when tm == nil?

I always tend to prefer null object https://github.com/tendermint/tendermint/blob/master/types/nop_event_bus.go in such cases or better not have this nil branch at all.

Reply:
Input validation is a bit of a habit for me, but I can remove the check if you wish.